really-core/src/main/scala/io/really/gorilla/ObjectSubscriber.scala

```diff
@@ -0,0 +1,129 @@
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+import akka.pattern.{ AskTimeoutException, ask, pipe }
+import io.really.model.Model
+import io.really.protocol.SubscriptionFailure
+import io.really.protocol.SubscriptionFailure.SubscriptionFailureWrites
+
+class ObjectSubscriber(rSubscription: RSubscription, globals: ReallyGlobals) extends Actor with ActorLogging
+    with Stash {
+
+  val r = rSubscription.r
+  val logTag = s"ObjectSubscriber ${rSubscription.pushChannel.path}$$$r"
+
+  var fields = rSubscription.fields.getOrElse(Set.empty)
+
+  val shotgun = context.system.scheduler.scheduleOnce(globals.config.GorillaConfig.waitForReplayer, self, PoisonPill)
+
+  def subscriptionFailed(errorCode: Int, reason: String) = {
+    log.error(s"$logTag is going to die since the subscription failed because of: $reason\n error code: $errorCode")
+    rSubscription.pushChannel ! SubscriptionFailureWrites.writes(SubscriptionFailure(r, errorCode, "Internal Server Error"))
+    context.stop(self)
+  }
+
+  def commonHandler: Receive = {
+    case Unsubscribe =>
+      log.debug(s"$logTag is going to die since it got an unsubscribe request")
+      rSubscription.requestDelegate ! Unsubscribe
+      context.stop(self)
+    case SubscriptionFailure(r, errorCode, reason) =>
+      subscriptionFailed(errorCode, "Internal Server Error")
+    case Terminated(actor) =>
+      subscriptionFailed(505, "Associated replayer stopped")
+  }
+
+  def receive: Receive = commonHandler orElse waitingModel
+
+  def waitingModel: Receive = {
+    case ReplayerSubscribed(replayer) =>
+      context.watch(replayer)
+      implicit val timeout = Timeout(globals.config.GorillaConfig.waitForModel)
+      val f = (globals.modelRegistry ? GetModel(rSubscription.r, self)).mapTo[ModelResult]
+      f.recoverWith {
+        case e: AskTimeoutException =>
+          log.debug(s"$logTag timed out waiting for the model object")
+          Future successful ModelResult.ModelFetchError(r, s"Request to fetch the model timed out for R: $r")
+        case NonFatal(e) =>
+          log.error(e, s"$logTag got an unexpected error while getting the model instance")
+          Future successful ModelResult.ModelFetchError(r, s"Request to fetch the model failed for R: $r with error: $e")
+      } pipeTo self
+      shotgun.cancel()
+    case evt @ ModelResult.ModelObject(m, _) =>
+      if (fields.isEmpty) {
+        fields = evt.model.fields.keySet
+      }
+      unstashAll()
+      context.become(withModel(m) orElse commonHandler)
+    case ModelResult.ModelNotFound =>
+      subscriptionFailed(503, s"$logTag couldn't find the model for r: $r")
+    case ModelFetchError(r, reason) =>
+      subscriptionFailed(504, s"$logTag couldn't fetch the model for r: $r because of: $reason")
+    case _ =>
+      stash()
+  }
+
+  def withModel(model: Model): Receive = {
+    case entry: GorillaLogCreatedEntry =>
```

Comment:
I think we aren't going to need that in the future and for the API consistency. I can discuss that with @AhmedSoliman again. you can't use one replayer to replay collection events and object events. In the future you will need two types of replayers. One will replay only `created` events and another one for object events (update and delete). So don't write code that you aren't sure that you will need that.